packages/mongo-nav/src/MongoNav.spec.tsx

```diff
@@ -1,68 +1,310 @@
-
-  const { getByText, getByTestId } = render(
-    <MongoNav
-      activeProduct="realm"
-      mode="dev"
-      onOrganizationChange={onOrganizationChange}
-      onProjectChange={onProjectChange}
-      urls={{
-        orgNav: {
-          accessManager: 'https://cloud.mongodb.com/access-manager-test',
-        },
-      }}
-      hosts={{ cloud: cloudHost }}
-    />,
-  );
+// types
+interface ExpectedElements {
+  [key: string]: nullableElement;
+}
+
+describe('packages/mongo-nav', () => {
+  const queries: Queries = {};
+  const expectedElements: ExpectedElements = {};
+
+  const setQueries = ({ queryByTestId }: Queries) => {
+    Object.assign(queries, { queryByTestId });
+    setExpectedElements();
+  };
+
+  const setExpectedElements = () => {
+    const { queryByTestId = () => null } = queries;
+    expectedElements.orgNav = queryByTestId('organization-nav');
+    expectedElements.projectNav = queryByTestId('project-nav');
+    expectedElements.admin = queryByTestId('org-nav-admin-link');
+    expectedElements.billing = queryByTestId('org-nav-billing');
+    expectedElements.activityFeed = queryByTestId('project-nav-activity-feed');
+    expectedElements.currentOrg = queryByTestId('org-select-active-org');
+    expectedElements.currentProject = queryByTestId(
+      'project-select-active-project',
+    );
+    expectedElements.userMenu = queryByTestId('user-menu-trigger');
+  };
+
+  let onOrganizationChange: jest.Mock;
+  let onProjectChange: jest.Mock;
+  let fetchMock: jest.Mock;
+
+  beforeEach(() => {
+    onOrganizationChange = jest.fn();
+    onProjectChange = jest.fn();
+    fetchMock = jest.fn();
+    window.fetch = fetchMock;
+  });
+
+  afterEach(() => {
+    delete window.fetch;
+    jest.restoreAllMocks();
+    cleanup();
+  });
+
+  const renderComponent = (props = {}) => {
+    setQueries(
+      render(
+        <MongoNav
+          activeProduct="cloud"
+          onOrganizationChange={onOrganizationChange}
+          onProjectChange={onProjectChange}
+          {...props}
+        />,
+      ),
+    );
+  };
 
   describe('by default', () => {
-    test('it renders the ProjectNav', () => {
-      const projectNav = getByTestId('project-nav');
-      expect(projectNav).toBeInTheDocument();
+    const responseObject = {
+      ok: true,
+      json: () => Promise.resolve(dataFixtures),
+    };
+
+    beforeEach(async () => {
+      fetchMock.mockResolvedValue(responseObject);
+      renderComponent();
+
+      await act(() => responseObject.json());
     });
 
-    test('it renders admin as false', () => {
-      const orgNav = getByTestId('organization-nav');
-      expect(orgNav.innerHTML.includes('Admin')).toBe(false);
+    test('fetch is called', () => {
+      expect(fetchMock).toHaveBeenCalledTimes(1);
+    });
+
+    test('the organization nav is rendered', () => {
+      const organizationNav = expectedElements['orgNav'];
+      expect(organizationNav).toBeInTheDocument();
```

Comment:
Why so much indirection? Why not just:

```suggestion
      expect(expectedElements.orgNav).toBeInTheDocument();
```

Ditto for other methods. 

I'd also just combine this and the next test to assert that the dom contains the elements you expect. No value added having them be separate.

Reply:
👍